bridge/setu/listener/rootchain.go

```diff
@@ -197,8 +211,12 @@ func (rl *RootChainListener) queryAndBroadcastEvents(fromBlock *big.Int, toBlock
 					if err := helper.UnpackLog(rl.stakingInfoAbi, event, selectedEvent.Name, &vLog); err != nil {
 						rl.Logger.Error("Error while parsing event", "name", selectedEvent.Name, "error", err)
 					}
-					if bytes.Compare(event.NewSigner.Bytes(), helper.GetAddress()) == 0 {
+					if bytes.Compare(event.NewSignerPubkey, helper.GetPubKey().Bytes()) == 0 {
```

Comment:
Event's public-key won't have `0x04` prefix while `helper.GetPubkey()` has a prefix. So it won't match directly. Also added some comments to add `signer` in the event: https://github.com/maticnetwork/contracts/pull/216#pullrequestreview-385289830

You might have to change ABI for the same.

Reply:
oh yeah....Great catch. thank you.